pkg/api/norman/customization/kontainerdriver/actionhandler.go

```diff
@@ -154,30 +156,40 @@ func (lh ListHandler) LinkHandler(apiContext *types.APIContext, next types.Reque
 		rkeSysImages[k8sVersion] = *rkeSysImgCopy
 	}
 
-	// get system charts path
-	systemCatalog, err := lh.CatalogLister.Get("", utils.SystemLibraryName)
+	var catalogImageList *v1.ConfigMap
+	catalogImageList, err = lh.ConfigMapLister.Get(utils.GetCatalogImageCacheName(utils.SystemLibraryName), metav1.GetOptions{})
 	if err != nil {
-		return httperror.WrapAPIError(err, httperror.ServerError, "error getting system catalog")
+		return
 	}
-	systemCatalogHash := helmlib.CatalogSHA256Hash(systemCatalog)
-	systemCatalogChartPath := filepath.Join(helmlib.CatalogCache, systemCatalogHash)
 
-	var targetImages []string
+	var targetRkeSysImages []string
+	var targetSysCatalogImages []string
 	switch apiContext.ID {
 	case linuxImages:
-		targetImages, _, err = image.GetImages(systemCatalogChartPath, "", nil, []string{}, rkeSysImages, image.Linux)
+		targetRkeSysImages, _, err = image.GetImages("", "", nil, []string{}, rkeSysImages, image.Linux)
+		if err != nil {
+			return httperror.WrapAPIError(err, httperror.ServerError, "error getting image list for linux platform")
+		}
+		_, targetSysCatalogImages, err = image.ParseCatalogImageListConfigMap(catalogImageList)
 		if err != nil {
 			return httperror.WrapAPIError(err, httperror.ServerError, "error getting image list for linux platform")
 		}
 	case windowsImages:
-		targetImages, _, err = image.GetImages(systemCatalogChartPath, "", nil, []string{}, rkeSysImages, image.Windows)
+		targetRkeSysImages, _, err = image.GetImages("", "", nil, []string{}, rkeSysImages, image.Windows)
 		if err != nil {
 			return httperror.WrapAPIError(err, httperror.ServerError, "error getting image list for windows platform")
 		}
+		targetSysCatalogImages, _, err = image.ParseCatalogImageListConfigMap(catalogImageList)
+		if err != nil {
+			return httperror.WrapAPIError(err, httperror.ServerError, "error getting image list for linux platform")
+		}
 	}
 
+	var targetImages []string
 	agentImage := settings.AgentImage.Get()
 	targetImages = append(targetImages, img.Mirror(agentImage))
+	targetImages = append(targetImages, targetRkeSysImages...)
```

Comment:
Same, not sure if targetRkeSysImages will have anything on it since empty path is passed.